File: Program/cinemmax/src/backend/classes/DTO/ShowTimeDetailsDTO/show-time-details-dto.ts
import { CinemaDTO } from "../CinemaDTO/cinema-dto";
import { FilmDTO } from "../FilmDTO/film-dto";
import { HallDTO } from "../HallDTO/hall-dto";
import { PricingDTO } from "../PricingDTO/pricing-dto";
import { ShowTypeDTO } from "../ShowTypeDTO/show-type-dto";

export class ShowTimeDetailsDTO {
    constructor(
        public id: number,
        public showTime: string,
        public film: FilmDTO,
        public showType: ShowTypeDTO,
        public hall: HallDTO,
        public cinema: CinemaDTO,
        public pricing: PricingDTO
    ) {}
}

File: Program/cinemmaxBackend/src/main/java/cinemmaxbackend/general/classes/DTO/ShowTimeDetails/ShowTimeDetailsDTO.java
package cinemmaxbackend.general.classes.DTO.ShowTimeDetails;

import cinemmaxbackend.general.classes.DTO.ShowType.ShowTypeDTO;
import cinemmaxbackend.general.classes.DTO.cinema.CinemaDTO;
import cinemmaxbackend.general.classes.DTO.film.FilmDTO;
import cinemmaxbackend.general.classes.DTO.hall.HallDTO;
import cinemmaxbackend.general.classes.DTO.pricing.PricingDTO;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@AllArgsConstructor
public class ShowTimeDetailsDTO {
    private Long id;
    private LocalDateTime showTime;
    private FilmDTO film;
    private ShowTypeDTO showType;
    private HallDTO hall;
    private CinemaDTO cinema;
    private PricingDTO pricing;
}
